src/util/helpers.py

```diff
@@ -214,3 +215,104 @@ def _get_theme_models():
         "theme.website.menu": "website.menu",
         "theme.ir.attachment": "ir.attachment",
     }
+
+
+FieldsPathPart = namedtuple("FieldsPathPart", "field_model field_name relation_model")
+FieldsPathPart.__doc__ = """
+Encapsulate information about a field within a fields path.
+
+:param str field_model: model of the field
+:param str field_name: name of the field
+:param str relation_model: target model of the field, if relational, otherwise ``None``
+"""
+for _f in FieldsPathPart._fields:
+    getattr(FieldsPathPart, _f).__doc__ = None
+
+
+def resolve_model_fields_path(cr, model, path):
+    """
+    Resolve model fields paths.
+
+    This function returns a list of :class:`~odoo.upgrade.util.helpers.FieldsPathPart`
+    where each item describes a field in ``path`` (in the same order). The returned list
+    could be shorter than the original ``path`` due to a missing field or model, or
+    because there is a non-relational field in the path. The only non-relational field
+    allowed in a fields path is the last one, in which case the returned list has the same
+    length as the input ``path``.
+
+    .. example::
+
+       To get the information about the fields path ``user_ids.partner_id.title_id``
+       from model `res.users`, we can call this function as
+
+       .. code-block:: python
+
+          resolve_model_fields_path(cr, "res.users", "user_ids.partner_id.title_id".split("."))
```

Comment:
Done, with better examples.

Reply:
I rebuilt the docs locally to be sure this is publish-ready:

![image](https://github.com/odoo/upgrade-util/assets/4387571/f183e122-d828-4038-9185-da0abb2353e8)

![image](https://github.com/odoo/upgrade-util/assets/4387571/d8264ab3-ef7b-450c-b6ba-0982df73c2cf)

Note, that it isn't built in the online docs because we do not publish, yet, the `helpers` subpackage.

cc: @andreabak